extensions/binding/org.eclipse.smarthome.binding.mqttbroker.test/src/test/java/org/eclipse/smarthome/binding/mqttbroker/UtilsTest.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.mqttbroker;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.eclipse.smarthome.binding.mqttbroker.internal.HexUtils;
+import org.junit.Test;
+
+/**
+ * Tests cases for {@link HexUtils}.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class UtilsTest {
+    @Test
+    public void validHex() {
+        final String hexTest = "a2a1a0";
+        final byte[] byteTest = { (byte) 162, (byte) 161, (byte) 160 };
+        byte[] test = HexUtils.fromHex(hexTest);
+        assertThat(test, is(byteTest));
+        assertThat(hexTest, is(HexUtils.toHexString(test)));
+    }
+
+    @Test
+    public void validHexWithColons() {
+        final String hexTest = "a2:a1:a0";
+        final byte[] byteTest = { (byte) 162, (byte) 161, (byte) 160 };
+        assertThat(HexUtils.fromHex(hexTest), is(byteTest));
+    }
+
+    @Test
+    public void validHexWithWhitespace() {
+        final String hexTest = "a2 a1 a0";
+        final byte[] byteTest = { (byte) 162, (byte) 161, (byte) 160 };
+        assertThat(HexUtils.fromHex(hexTest), is(byteTest));
+    }
+
+    @Test
+    public void validHexUppercase() {
+        final String hexTest = "A2 A1 A0";
+        final byte[] byteTest = { (byte) 162, (byte) 161, (byte) 160 };
+        assertThat(HexUtils.fromHex(hexTest), is(byteTest));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void invalidHex() {
+        final String hexTest = "a2a1a0ll";
+        byte[] test = HexUtils.fromHex(hexTest);
+        assertThat(hexTest, is(HexUtils.toHexString(test)));
```

Comment:
In line 53 (`@Test(expected = IllegalArgumentException.class)`) you expect the code to throw the IAE which is done by the code in line 56. So I guess line 57 is never reached.

The following test case will run happily:

```
@Test(expected = IllegalArgumentException.class)
public void invalidHex() {
    final String hexTest = "a2a1a0ll";
    HexUtils.fromHex(hexTest);
    assertThat(true, is(false)); // should fail here if the line is reached.
}
```